partner/WHMCS/modules/servers/jetpack/jetpack.php

```diff
@@ -0,0 +1,288 @@
+}
+
+
+/**
+ * A WHMCS module for use by Jetpack hosting partners to provision Jetpack plans.
+ * The module provides functionality for partner hosts to be able to save their
+ * client id and secret to request an access token for provisioning plans.
+ *
+ * Plans available for provisioning include free, personal, premium and professional
+ *
+ * A host has options to either provision(Create == WHMCS equivalent functional term)
+ * or Cancel(Terminate == WHMCS equivalent functional term) from the WHMCS client area.
+ *
+ * Host setup for custom fields is currently required in order to use the module.
+ *
+ */
+
+/**
+ * Jetpack Meta Data for WHMCS module.
+ * @return array
+ */
+function jetpack_MetaData()
+{
+    return array(
+        'DisplayName' => 'Jetpack by Automattic',
+        'Description' => 'Use this module to provision Jetpack plans with your Jetpack hosting partner account',
+        'APIVersion' => '1.1',
+        'RequiresServer' => false,
+    );
+}
+
+
+/**
+ * Basic configuration options required for a partner to get
+ * a Jetpack plan provisioned. Currently a partner client id
+ * and secret are the only host partner options needed to get
+ * an access token to provision a Jetpack plan
+ * @return array
+ */
+function jetpack_ConfigOptions()
+{
+    return array(
+        'Jetpack Partner Client ID' => array(
+            'Type' => 'text',
+            'Size' => '256',
+        ),
+        'Jetpack Partner Client Secret' => array(
+            'Type' => 'text',
+            'Size' => '256',
+        )
+    );
+}
+
+
+/**
+ * Equivalent to /provision. Create a Jetpack plan using
+ * a Jetpack Hosting partner account.
+ *
+ *
+ * @param array $params
+ * @return string 'success'
+ * @throws Exception
+
+ */
+function jetpack_CreateAccount(array $params)
+{
+
+    $validation_status = validate_required_fields($params);
+    if ($validation_status !== true) {
+        return $validation_status;
+    }
+
+    try {
+        $access_token = get_access_token($params);
+        $provisioning_url = "https://public-api.wordpress.com/rest/v1.3/jpphp/provision";
+        $stripped_url = preg_replace("(^https?://)", "", $params['customfields']['Site URL']);
+
+        $request_data = array (
+            'plan' => strtolower($params['customfields']['Plan']),
+            'siteurl' => $stripped_url,
+            'local_user' => $params['customfields']['Local User'],
+            'force_register' => true,
+        );
+
+        $response = make_api_request($provisioning_url, $access_token, $request_data);
+        if ($response->success && $response->success == true) {
+            if ($response->next_url) {
+                save_provisioning_details($response->next_url, $params);
+            } elseif (!$response->next_url && $response->auth_required) {
+                save_provisioning_details($response->next_url, $params, true);
+            }
+
+            return 'success';
+        } else {
+            return $response;
```

Comment:
Wouldn't `$response` be an object here? Is it OK to return an object, instead of a string?

Reply:
@nb the make_api_request function was previously trying to handle the response object and return an error string if the response wasn't 2xx which would have technically caught this. In hindsight this isn't great and actually really confusing in the state it is where you commented on things as $response could have either been an object or a string. make_api_request no longer does this and will always return an object and it's up to the calling function to handle things from there.